openedx/features/course_duration_limits/resolvers.py

```diff
@@ -164,8 +164,7 @@ def get_template_context(self, user, user_schedules):
 
 def _get_verified_upgrade_link(user, schedule):
     enrollment = schedule.enrollment
-    if verified_upgrade_link_is_valid(enrollment):
```

Comment:
what is this line doing?

Reply:
https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/date_summary.py#L408-L429 is the original source.

The problem is that it's checking `enrollment.upgrade_deadline`, which comes from the schedule, but we don't actually want to check that in this case. It's also checking `enrollment.mode in CourseMode.UPSELL_TO_VERIFIED_MODES`, but that's covered by the code only selecting enrollments where `course_duration_limits` are enabled.